### ISSUE TITLE
Modal dialog/window

### ISSUE THREAD

**Issue by fuintis** [fyne-io/fyne#291]
Is there a way to block the parent window when a dialog (or child window) is shown?

**Comment by andydotxyz** [comment]
We have recently added a modal overlay which greys out the window and positions the content within but over the top. Would this be what you were thinking of?
I was considering moving dialogs to use this instead as we don't have a cross-platform way to force windows to block in that way.

**Comment by fuintis** [comment]
> modal overlay which greys out the window and positions the content within but over the top

Just like html5 does.

> Would this be what you were thinking of?

Yes, something like that.

> We have recently added a modal overlay

How can i use it?


Many thanks for your help.

**Comment by andydotxyz** [comment]
It will be part of the 1.1 release which we expect in a couple of weeks.
It's probably easiest to wait for release if you can?

**Comment by fuintis** [comment]
Yes, probably.

I'll wait to the next release, then.

Many thx!

**Comment by mar1ged** [comment]
Sorry for reviving this old issue.

I came here in search for a GUI library for go that allows me to display a dialog which can't be ignored. I need to write a go program that sits in the background and waits for a certain time. When the point in time is reached the program needs to display a dialog with some text in it. This dialog needs to be the topmost window. Other windows can still be visible in the background but my dialog must not be put into the background (at least until the `OK` button is clicked.

I ran your fyne_demo and very much like what I see. When using the 1.1 version will I be able to create such a dialog, in my case on Windows ?

**Comment by andydotxyz** [comment]
You can use Window.RequestFocus() to ask any window to raise, this is just a request, however, and the operating system may choose to ignore it depending on circumstances.

We don't support a "force on top" feature at this time. Also note that our dialogs currently require a parent window so you should use a regular window instead.

**Comment by mar1ged** [comment]
Thanks for your suggestion. I modified the `fyne_demo` and added `w.RequestFocus()` to the centered windows test. Unfortunately the newly opened window is neither on top of for example `calc` nor on top of its parent.
So I think I need to look for a different solution.

![grafik](https://user-images.githubusercontent.com/16045549/65624031-58a2de00-dfc9-11e9-8da9-33715f7077ed.png)

**Comment by andydotxyz** [comment]
That sounds like a bug - a new window should always appear above its parent unless you have some strange configuration in your OS?...

**Comment by mar1ged** [comment]
Sorry for the confusion: it comes up on top but I can position/move it like shown on the screen

**Comment by okratitan** [comment]
Your application has no rights to control over other windows from outside your application - therefore it can not control stacking, block, etc... it is the job of the desktop/window manager to control this - and for good reasons that I won't get into. So if that's the case you may be asking well then how does it work? Well every platform will be different. On Linux we would send hints to the window manager to request that our window be managed in this way.  I'm sure other platforms like Mac and Windows will be similar.

Long story short - Fynr would have to expose an API to you to make this possible. And then Fynr would have to handle the backend native code to set up these hints.  This can be problematic and a chore unless we can get the exact behavior to work cross platform. Else we will have applications having inconsistent behavior. We can look at doing this but no promises!

**Comment by andydotxyz** [comment]
As I mentioned above @mar1ged we don’t support the always on top hints. As @okratitan points out this could be difficult. That said it’s not something that can be supported on all platforms (i.e. on iOS there is no such thing) so should we add it at all?

**Comment by mar1ged** [comment]
I am fine (fyne ? ;-)) with that, just wanted to make sure I don't miss something. I understand that a generic solution can't support all platform specific stuff

**Comment by TudorHulban** [comment]
I believe Lazarus with Object Pascal supports modal cross platform.
